src/sherpa_ai/database/user_usage_tracker.py

```diff
@@ -69,17 +77,44 @@ def get_whitelist_by_user_id(self, user_id):
 
     def is_in_whitelist(self, user_id):
         return bool(self.get_whitelist_by_user_id(user_id))
+    
+    def add_and_check_data(self, combined_id, token, reset_timestamp=False , reminded_timestamp=False):
+        self.add_data(combined_id=combined_id , token=token, reset_timestamp=reset_timestamp, reminded_timestamp=reminded_timestamp)
+        self.token_left_calculator(combined_id=combined_id)
 
-    def add_data(self, combined_id, token, reset_timestamp=False):
+    def add_data(self, combined_id, token, reset_timestamp=False , reminded_timestamp=False):
         data = UsageTracker(
             user_id=combined_id,
             token=token,
             timestamp=int(time.time()),
             reset_timestamp=reset_timestamp,
+            reminded_timestamp=reminded_timestamp
         )
         self.session.add(data)
         self.session.commit()
 
+    def percentage_used(self , combined_id):
+        total_token_since_last_reset = self.get_sum_of_tokens_since_last_reset(
+                user_id=combined_id
+            )
+        return  (total_token_since_last_reset*100)/self.max_daily_token 
+
+
+    def token_left_calculator(self, combined_id):
+        split_parts = combined_id.split('_')
+        user_id = ""
+        if len(split_parts) > 0:
+            user_id = split_parts[0]
+
+        user_is_whitelisted = self.is_in_whitelist(user_id)
+        self.is_reminded = self.check_if_reminded(combined_id=combined_id)
+        if not user_is_whitelisted and not self.is_reminded:
+            if self.percentage_used(combined_id=combined_id) > 75 and not self.is_reminded:
```

Comment:
75 is a "magic number" here. Define it as a constant at the top of the code, with a name suggesting what it is. e.g. WARNING_THRESHOLD_PCT.